src/table/table-role/__tests__/grid-navigation-context.test.tsx

```diff
@@ -0,0 +1,72 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useCallback, useRef } from 'react';
+import { render } from '@testing-library/react';
+import { GridNavigationContext } from '../../../../lib/components/table/table-role/grid-navigation-context';
+import { useGridNavigationFocusable } from '../../../../lib/components/table/table-role';
+import { renderWithGridNavigation } from './utils';
+
+function Button(props: React.HTMLAttributes<HTMLButtonElement>) {
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const { tabIndex } = useGridNavigationFocusable(buttonRef, { tabIndex: props.tabIndex });
+  return <button {...props} ref={buttonRef} tabIndex={tabIndex} />;
+}
+
+function ButtonAlt(props: React.HTMLAttributes<HTMLButtonElement>) {
+  const getElement = useCallback(() => document.querySelector(`#${props.id}`), [props.id]);
+  const { tabIndex } = useGridNavigationFocusable(getElement, { tabIndex: props.tabIndex });
+  return <button {...props} tabIndex={tabIndex} />;
+}
+
+test.each([Button, ButtonAlt])('does not override tab index when keyboard navigation is not active', Button => {
```

Comment:
Will this produce two test cases with the exact same name? Or is jest smart enough to somehow add the name of the Button component into the test case?

I'm asking because imagine a scenario where this test fails but only for `ButtonAlt`. Would you be able to tell from the Jest output which button variant failed?

Reply:
Makes sense, I will think on how to modify the name depending on the selected case